lib/Sema/CSBindings.cpp

```diff
@@ -751,3 +751,101 @@ ConstraintSystem::getPotentialBindings(TypeVariableType *typeVar) {
 
   return result;
 }
+
+/// \brief Enumerates all of the 'direct' supertypes of the given type.
+///
+/// The direct supertype S of a type T is a supertype of T (e.g., T < S)
+/// such that there is no type U where T < U and U < S.
+static SmallVector<Type, 4> enumerateDirectSupertypes(TypeChecker &tc,
+                                                      Type type) {
+  SmallVector<Type, 4> result;
+
+  if (type->mayHaveSuperclass()) {
+    // FIXME: Can also weaken to the set of protocol constraints, but only
+    // if there are any protocols that the type conforms to but the superclass
+    // does not.
+
+    // If there is a superclass, it is a direct supertype.
+    if (auto superclass = tc.getSuperClassOf(type))
```

Comment:
I think `getSuperClassOf()` should be replaced with just calling `type->getSuperclass()` -- can you see if that works here and elsewhere?

Reply:
That should indeed work because `PotentialBinding` strips away parens anyway, and there are no other users of this function.